npm/src/compiler/events.ts

```diff
@@ -53,37 +35,47 @@ function makeResultObj(): ShotResult {
   return { success: false, result: "", events: [] };
 }
 
-export class QscEventTarget extends EventTarget implements IQscEventTarget {
+// The actual event target implementation adds one more event type
+// to notify the UI that the results should be refreshed.
+// This event does not come from the compiler service itself
+// so it's not exposed as part of QscEvents or IQscEventTarget.
+// Direct consumers of QscEventTarget can attach a listener for
+// this event.
+type QscUiEvents =
+  | QscEvents
+  | (Event & { type: "uiResultsRefresh"; detail: undefined });
+
+export class QscEventTarget implements IQscEventTarget {
+  private eventTarget = new EventTarget();
```

Comment:
Was there a reason you changed it from deriving from EventTarget to aggregating one?

Reply:
Yes. TypeScript complains if you derive from `EventTarget`. When you derive from a class you need to also implement all its methods with their original signature.

`EventTarget` has `addEventListener(type: string, callback: EventListenerOrEventListenerObject | null, options?: AddEventListenerOptions | boolean): void;` which is more general than what we have. If you implement `addEventListener`  then you need to be able to handle it if someone passes in `EventListenerOrEventListenerObject` for the `callback` parameter. Which we don't.

[playground example showing error](https://www.typescriptlang.org/play?#code/C4TwDgpgBAigzgYwKIDcIDthygXiqjYKAMigG8BYAKClqlEgC4oAiAWwjjgEMBzCFgBpqdKABMIwbgEsANszjAATtPS9qAXygAfciLoMIzFmICubMEP20JUucwD2AIwBWEBME3Vqq4BCUAZtwI0ACS8MhomAAq3Er8RJQ0dNxiYgSYADLSihj+ADzRUBAAHn7oYtgRGVgA2iyGLAC6AHwAFNai9OBGUNHCyV1Qsjnl-sxtbRBRwMw1JPhlSsHA+dUzcILk3Ux9UBotAJS4LVAoDtJix7roprKynYfM55cA3N6DUEoQbA5oNdlcugCkVSuVKrBEDU4PVGq0Op9RIZmP1OqIRkDxlBJtNCHMZgskEsVmsoRstmQdr0igdjjhTi8rjooLd7o9nhcxO8qJ0xDkwNxgAgABY1KYzfGEBbrQhwJ5QJwOByyCDcdDcjQfBCybhcSGRQixeKSADiSrEUGkFhVHEw2HCZMNcQSek+qXSM0BYyUhWKZQwEJldthPWa7TRBh6KIGQ1oGO9E3FePwBNIROUJKDWApVJR+yOJzOnOuLLuD0+xzImp5n2+v3+ntGeR9oP9FSqjuDDVD8IjtGRfRjsfjzcTuMwkswhOJHlJBrtOYHNIL9KLlxLrPLokr1d5-MFIrF49mKalpCzcu232ApiU6CgQVkcGg1d3VG1uo78+ARoSACFUj9cFsBqX9JEta0fllKAHW-MDEk6d0ASbYEWyAgMv2hENIDDBFY1zQc+2GFCsRxCVTynNMZ1WC9FyjPZaULRkNzLR5yDfUQ6z+CBkMxNCwQw-UsO7HDe0RSNdlRcS4xIpQx3I+YqIzWdaO2Jd8zpBli2ZTc2KrD5RD5OABSFUUZiTCcKKIc9OywSsvkkW970fZ99i8GsAHoPJggJxH8aQ0CgbgqQfJQHDYKz4K2UIoDgYUHDuC0nGgbgnBVegHCgAU9WkIhfEy4BhRyB9THQDxpAcdBqACUrysq+huAAa04UDnUkNoT1a41gErTpgAAOiQxs+LaFgAiVIRS3uQ5NCAA)

I have no clue why TypeScript liked it the way you originally had it, but I couldn't get around the warning.